File: 3952-trionic-array-i/trionic-array-i.java
class Solution 
{
    public boolean isTrionic(int[] nums) 
    {
        int pos1=0,pos2=0,pos3=nums.length;

        for(int i=1;i<nums.length;i++)
        {
            if(nums[i]<=nums[i-1] || i==nums.length-1)
            {
                pos1=i-1;
                break;
            }
        }

        if(pos1==0)
            return false;

        for(int i=pos1+1;i<nums.length;i++)
        {
            if(nums[i]>=nums[i-1] || i==nums.length-1)
            {
                pos2=i-1;
                break;
            }
        }

        for(int i=pos2+1;i<nums.length;i++)
        {
            if(nums[i]<=nums[i-1])
            {
                pos3=i;
                break;
            }
        }

        if(pos1==pos2 || pos2==pos3 || pos3!=nums.length)
            return false;

        return true;        
    }
}

File: 3952-trionic-array-i/trionic-array-i.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) 
    {
        int n = nums.size();
        int i = 1;

        
        while (i < n && nums[i] > nums[i - 1]) 
        {
            i++;
        }
        int p = i - 1;
        if (p == 0) 
           return false;  
        if (i == n) return false;

        
        while (i < n && nums[i] < nums[i - 1]) 
        {
            i++;
        }
        int q = i - 1;
        if (q == p) 
           return false;  
        if (i == n) return false;

        
        while (i < n && nums[i] > nums[i - 1]) 
        {
            i++;
        }

        return i == n;
    }
};